File: task4/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	xValues := []float64{0.2, 0.3, 0.38, 0.43, 0.57}

	fmt.Println("x\t\ty")
	fmt.Println("------------------------")

	for i, x := range xValues {
		y := (math.Pow(math.Sin(x), 3) + math.Pow(math.Cos(x), 3)) * math.Log(x)
		fmt.Printf("x%d = %.2f\t%.6f\n", i+1, x, y)
	}
}

File: task4/main.ts
const xn = 0.11;
const xk = 0.36;
const step = 0.05;

function calcYnew(x: number): number {
  return (Math.sin(x) ** 3 + Math.cos(x) ** 3) * Math.log(x);
}

console.log("x\t\ty");
console.log("------------------------");

for (let x = xn; x <= xk + 1e-9; x += step) {
  const y = calcYnew(x);
  console.log(`${x.toFixed(2)}\t\t${y.toFixed(6)}`);
}
